src/Symfony/Component/ErrorHandler/ErrorHandler.php

```diff
@@ -153,6 +153,32 @@ public static function register(self $handler = null, bool $replace = true): sel
         return $handler;
     }
 
+    /**
+     * Calls a function and turns any PHP warnings into \ErrorException.
+     *
+     * @return mixed Returns $function(...$arguments)
+     *
+     * @throws \ErrorException When $function(...$arguments) triggers a PHP warning
+     */
+    public static function call(callable $function, ...$arguments)
+    {
+        set_error_handler(static function (int $type, string $message, string $file, int $line) {
```

Comment:
> @throws \ErrorException When $function(...$arguments) triggers a PHP warning

I don't see any check on the error type.

Reply:
we shouldn't add one so fit the comment - but in practice, only warnings are used by the engine for the cases we plan to catch
so I'm not sure we should change anything
what would be your proposal?

Reply:
This feature can technically be misused and other types of error can end in this temporary error handler. We can be defensive and check the error type. If !== than E_WARNING, we call the previous handler. Or we can just say that this method handles all error types.

Reply:
I think the point of this method is to opt-out from PHP warnings/notices/whatever, so I would not add the check. Now, do we need to enumerate all error codes and be that accurate?

Reply:
In this case let's just change the feat to : "added `ErrorHandler::call()` method utility to turn any PHP **error** into `\ErrorException`"

Reply:
Another solution is to add a basic method:

```php
public static function callWithLevel(int $level, callable $func, ...$args)
{
    set_error_handler(..., $level);
}
```

It can be reused by `call { return callWithLevel(E_ALL, ...) }`

Reply:
> callWithLevel

no please, let's keep things simple

> method utility to turn any PHP error

except that 100% of the target use case is for php warnings :)

Reply:
changed `PHP warning` by `PHP error` everywhere